signer/src/storage/model.rs

```diff
@@ -1,18 +1,24 @@
 //! Database models for the signer.
 
+use fake::faker::time::en::DateTimeAfter;
+
 /// Bitcoin block.
 #[derive(Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "testing", derive(fake::Dummy))]
 pub struct BitcoinBlock {
     /// Block hash.
+    #[dummy(expr = "fake::vec![u8; 32]")]
```

Comment:
Ahh, good to know. I was wondering how to constrain the faker to always return 32 bytes without using `[u8; 32]`. I mistakenly assumed it wasn't possible.

Related, why not use `[u8; 32]` for the aliases where we want a fixed number of bytes?

Reply:
Yeah I think writing is fine because it takes slices, but reading doesn't support fixed-size arrays because sqlx can't guarantee the size of the read data.

> We could use postgres [domains](https://www.postgresql.org/docs/16/sql-createdomain.html) to get the fixed length "data type". I'm not trying to suggest that we do so and I do not have a good sense on whether doing so would be a net benefit, I just wanted to point out that it is possible.

Interesting, didn't know of this. Thanks for sharing! :pray: